lib/src/util/cast_ui_factory.dart

```diff
@@ -0,0 +1,20 @@
+import 'package:over_react/component_base.dart';
+
+/// Casts [value] to `UiFactory<T>`, which can be inferred from the surrounding context,
+/// allowing you to perform casts of dynamic values without having to repeat the type.
+///
+/// Example:
+///
+/// ```dart
+/// UiFactory<FooProps> Foo = castUiFactory(_$Foo);
```

Comment:
I think showing the alternative here would be helpful, as well as showing the ignore comments that are still necessary:
```suggestion
/// // Explicit cast
/// UiFactory<FooProps> Foo = 
///    // ignore: undefined_identifier
///    _$Foo as UiFactory<FooProps>;
///
/// // This utility
/// UiFactory<FooProps> Foo = castUiFactory(_$Foo); // ignore: undefined_identifier
```